spec.html

```diff
@@ -36083,6 +36093,12 @@ <h1>LeaveCriticalSection ( _WL_ )</h1>
         <p>The abstract operation LeaveCriticalSection takes one argument, a WaiterList _WL_. It performs the following steps:</p>
         <emu-alg>
           1. Assert: The calling agent is in the critical section for _WL_.
+          1. Let _execution_ be the [[CandidateExecution]] field of the calling surrounding's Agent Record.
+          1. Let _eventsRecord_ be the Agent Events Record in _execution_.[[EventsRecords]] whose [[AgentSignifier]] is AgentSignifier().
+          1. Let _leaverEventList_ be _eventsRecord_.[[EventList]].
+          1. Let _leaveEvent_ be a new Synchronize event.
+          1. Append _leaveEvent_ to _leaverEventList_.
+          1. Set the Synchronize event in _WL_ to _leaveEvent_.
```

Comment:
Should this be an internal slot, instead of a vague concept of being "in" a WaiterList?

Reply:
I don't recall exactly the history of why WaiterList is a "semantic object" instead of being a Record with fields. Perhaps it was to distinguish it as a special spec object that can be touched by different agents and needs to be locked by the critical section to be used? That way we can keep thinking of Records as objects that are intra-agent and don't need any extra care for use.

This also affects how I write the `Atomics.waitAsync` spec, since that extends the WaiterList to include more information that are accessed from multiple agents.

My preference is to keep it as a semantic object for this PR. I'm open to having some specialization of Record to distinguish WaiterList from other Records -- any thoughts there?

Reply:
I'm more confused by the whole concept of "semantic object" - the spec doesn't have anything else like that afaik.

It's fine to leave as-is, for the reasons you stated, but it seems like it'd be clearer to me in the future to have it be a proper Record; or to define some kind of more explicit syntax for interacting with "semantic objects" besides prose.